extensions/mssql/src/sqlToolsServer.ts

```diff
@@ -60,7 +60,7 @@ export class SqlToolsServer {
 			statusView.show();
 			statusView.text = 'Starting service';
 			this.client.start();
-			await this.activateFeatures(context);
+			await Promise.all([this.activateFeatures(context), clientReadyPromise]);
```

Comment:
is this going to delay loading other extensions during startup?  If so, I'd prefer to have a separate notification that the service is loaded that extensions can optionally subscribe to.  Basically, we should allow extensions that don't need the service loaded to begin activation as soon as possible.

Reply:
Yeah, fair. I wasn't sure how big of a problem that'd be but if you think it's a large enough concern I'll look into adding an event for this being ready that dependent extensions can hook in to.

Reply:
I think the service starting can take several seconds, and a bunch of extensions wait for mssql to finish activating before they start activating (azure, bdc, import, dacfx, etc.).  Many of these extensions also start services and we've gotten a lot of feedback that things initialize too slowly in general.

We'd want to look at actual perf numbers to really understand impact, but my intuition is that this would add another "long" blocking task on a critical path.

Reply:
I might also just make it so the command queues up requests if the service isn't activated yet and then sends any queued requests once the service is started. We'll see how simple it is to expose an event like my original thought (it seems like it shouldn't be too bad).

Reply:
I don't think this will have a impact on startup, currently the only extensions that depend on the mssql for activation is `[admin-tool-ext-win, agent, cms, dacpac, import, profiler]` and all these only have entry points through the mssql extension anyways (oe). So currently they all are currently waiting behind the mssql extension in a way. You can profile the extension host with this change and see if there is any difference.

On a side note, we have a lot of extensions with the activation event of `*` which isn't ideal and also impacts starting time.

Reply:
@anthonydresser @kburtram So I looked into this and here's some numbers I got from profiling start-up

| Counter        | Avg. Activate Time     
| ------------- |:-------------:|
| Await Client w/o Service Download      | 1,105ms |
| Don't Await Client w/o Service Download | 457ms |

So there is a pretty significant different in performance (as expected). Although it's definitely not that big of a hit - after all the app isn't unresponsive during this time. You're just limited in what you can do.

Anthony brings up a good point in that this extension is pretty much all about the tools service. Plus we're actually already waiting on the other features the extension provides (such as Credentials store and Azure Resource provider).

So while I generally think the pattern of allowing dependent extensions to subscribe to the features they're interested in is good in this case I don't really see that changing the overall performance enough - if we really wanted to tackle that there's other things we could do (such as having a shared Tools Service that doesn't need to be launched every startup) that would have a much bigger impact.

At this point then I'm leaning towards reverting back to iteration 1 where we just await the Tools Service being ready. @kburtram Are you OK with that and we can focus on more targeted improvements for affecting startup time when that's something we decide to work on?